api/handlers.go

```diff
@@ -1,219 +1,75 @@
-		WriteError(w, err, http.StatusInternalServerError)
-		return
-	}
-	if u.Tier == database.TierUnconfirmed {
-		// TODO This error should be picked up by the FE and it should offer to
-		// 	resend the account confirmation email.
-		WriteError(w, ErrAccountUnconfirmed, http.StatusUnauthorized)
-		return
-	}
-	err = u.VerifyPassword(pw)
-	if err != nil {
-		WriteError(w, errors.New("Bad username or password."), http.StatusUnauthorized)
-		return
-	}
-	token, err := IssueToken(u)
-	if err != nil {
-		fmt.Println(err)
-		// TODO WriteError doesn't set the response's error message properly. Or Postman doesn't read it properly?
-		WriteError(w, err, http.StatusUnprocessableEntity)
-		return
-	}
-	w.WriteHeader(http.StatusOK)
-	WriteJSON(w, token)
-}
-
-// passwordResetRequestHandler starts the password recovery routine.
-// This involves sending an email with a password reset link to the user.
-func (api *API) passwordResetRequestHandler(w http.ResponseWriter, req *http.Request, ps httprouter.Params) {
-	// TODO Implement
-	WriteJSON(w, struct{ msg string }{"Not implemented."})
-}
-
-// userPasswordResetVerifyHandler verifies that a password recovery request is
-// legitimate by verifying that the provided secret matches the sent one and
-// also that the code hasn't expired.
-func (api *API) userPasswordResetVerifyHandler(w http.ResponseWriter, req *http.Request, ps httprouter.Params) {
-	// TODO Implement
-	WriteJSON(w, struct{ msg string }{"Not implemented."})
-}
-
-// passwordResetCompleteHandler completes the password recovery routine by
-// changing the user's password to the newly provided one, given that the
-// provided recovery code is valid and unused. It also marks the recovery code
-// as used.
-func (api *API) passwordResetCompleteHandler(w http.ResponseWriter, req *http.Request, ps httprouter.Params) {
-	// TODO Implement
-	WriteJSON(w, struct{ msg string }{"Not implemented."})
-}
-
-// jwtToken is a helper function that extracts the JWT token from the context
-// and returns the contained user id, claims and the token itself.
-func jwtToken(req *http.Request) (id string, claims jwt.MapClaims, token *jwt.Token, err error) {
-	t, ok := req.Context().Value(ctxValue("token")).(*jwt.Token)
-	if !ok {
-		err = errors.New("failed to get token")
-		return
-	}
-	if reflect.ValueOf(t.Claims).Kind() != reflect.ValueOf(jwt.MapClaims{}).Kind() {
-		err = errors.New("the token does not contain the claims we expect")
-		return
-	}
-	claims = t.Claims.(jwt.MapClaims)
-	if reflect.ValueOf(claims["user_id"]).Kind() != reflect.String {
-		err = errors.New("the token does not contain the user_id we expect")
-	}
-	id = claims["user_id"].(string)
-	token = t
-	return
-}
+// TODO This will be needed bu only for changing the user's tier and expiration times. It will be driven by payments.
+//// userHandlerPUT updates an existing user.
+//func (api *API) userHandlerPUT(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
```

Comment:
If this is commented out, perhaps we should get rid of it in the documentation as well. Also I believe the documentation doesn't mention user tier and expiration time as fields that you could update on a user, but rather mentions first and last name plus email.